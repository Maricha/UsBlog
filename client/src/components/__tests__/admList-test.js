import React from 'react';
import { shallow } from 'enzyme'; 
import AdmItemList from '../admItemList';

describe('AdmList Component', () => {
  it('renders correctly', () => {
    const posts = [
      {
        id: 1,
        title: "title1"
      },
      {
        id: 2,
        title: "title2"
      }
    ]

    const component = shallow(<AdmItemList posts={posts} subscribeToMore={()=>{}}/>);
    expect(component).toMatchSnapshot();
  })
});