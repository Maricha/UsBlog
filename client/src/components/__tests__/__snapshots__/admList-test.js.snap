// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdmList Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AdmItemList)
    posts={
      Array [
        Object {
          "id": 1,
          "title": "title1",
        },
        Object {
          "id": 2,
          "title": "title2",
        },
      ]
    }
    subscribeToMore={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "itemWrapper": "AdmItemList-itemWrapper-2",
        "mainStyle": "AdmItemList-mainStyle-1",
      },
      "posts": Array [
        Object {
          "id": 1,
          "title": "title1",
        },
        Object {
          "id": 2,
          "title": "title2",
        },
      ],
      "subscribeToMore": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "itemWrapper": "AdmItemList-itemWrapper-2",
          "mainStyle": "AdmItemList-mainStyle-1",
        },
        "posts": Array [
          Object {
            "id": 1,
            "title": "title1",
          },
          Object {
            "id": 2,
            "title": "title2",
          },
        ],
        "subscribeToMore": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
